[messedup] Lua check adding and fixing

Alberth289346 opened the pull request:
Besides my 2 convenience fixes, this is a first test in how far we jump
I think I got most things that needed fixing

To consider
- [x] I copied the lucheckrc file from bob, which I didn't thoroughly review
- [x] Near the top of the patch queue it gets messy, that may need cleaning, or even re-shuffling all patches per file or something else that is easy to check
- [x] I skipped "unused argument Foo", which needs disabling, I think
- [x] There are some other remaining messages too, part of the submission is to see which ones, and it needs a decision what to do with them

TheCycoONE commented:
Eventually I'd like the commit to add luacheck to be at the end, so that they all pass travis in between.

TheCycoONE commented on CorsixTH/Lua/filesystem.lua:
```diff
@@ -18,7 +18,8 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 SOFTWARE. --]]
 
-local LFS = require "lfs"
+require "lfs"
```
Similarly, typically in lua if nothing is done with the require it's not necessary. (It's possible for the require to define globals, but that's considered bad practice.)

TheCycoONE commented on CorsixTH/Lua/game_ui.lua:
```diff
@@ -27,10 +27,8 @@ class "GameUI" (UI)
-local WM = SDL.wm
-local lfs = require "lfs"
-local pathsep = package.config:sub(1, 1)
+require "sdl"
```
Try without them

TheCycoONE commented on CorsixTH/Lua/graphics.lua:
```diff
@@ -19,7 +19,8 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 
 local TH = require "TH"
-local SDL = require "sdl"
+require "sdl"
```
broken record

TheCycoONE commented on CorsixTH/Lua/room.lua:
```diff
@@ -18,7 +18,7 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 SOFTWARE. --]]
 
-local TH = require "TH"
+require "TH"
```
""

TheCycoONE commented on CorsixTH/Lua/sprite_viewer.lua:
```diff
@@ -22,8 +22,9 @@ SOFTWARE. --]]
 -- when you don't have a copy of AnimView, or the means to compile it, then a
 -- crude sprite viewer is better than no sprite viewer.
 
+require"lfs"
```
""

TheCycoONE commented on CorsixTH/Lua/class.lua:
```diff
@@ -65,8 +65,8 @@ local function define_class(name, super, adopts_self)
 
-  local function new_class(methods, ...)
-    local constructor = methods[name]
+  local function new_class(methods2, ...)
```
Not a fan of calling it methods2, as that implies it's different from methods. It's not, but call on a metatable passes the actual table as the first argument. ```t``` is a fairly conventional name for that first argument. ```m``` would be fine as well in our case.

TheCycoONE commented on CorsixTH/Lua/dialogs/edit_room.lua:
```diff
@@ -1209,11 +1209,11 @@ function UIEditRoom:setDoorBlueprint(x, y, wall)
   if self.room_type.swing_doors then
-    for _, anim in ipairs(anim) do
-      anim:setAnimation(self.anims, 126, flags)
+    for _, anim2 in ipairs(anim) do
```
This function is huge. Suggest breaking it up, which might avoid using silly variable names.

TheCycoONE commented on CorsixTH/Lua/entities/patient.lua:
```diff
@@ -199,7 +199,7 @@ function Patient:completeDiagnosticStep(room)
     local attn_detail = room.staff_member.profile.attention_to_detail / divisor
     local skill = room.staff_member.profile.skill / divisor
-    local diagnosis_bonus = (attn_detail + 0.4) *  skill
+    diagnosis_bonus = (attn_detail + 0.4) *  skill
```
Two spaces between * and skill. I know you didn't put them there, but since you touched the line anyway, might as well fix it.

TheCycoONE commented on CorsixTH/Lua/entity_map.lua:
```diff
@@ -45,8 +45,8 @@ end
 function EntityMap:addEntity(x,y,entity)
-  local function add_entity_to_table(entity,entity_table)
-    entity_table[#entity_table+1] = entity
+  local function add_entity_to_table(enty, entity_table)
```
Promote the local function to the top level. It doesn't use the closure and that avoids renaming the variable.

TheCycoONE commented on CorsixTH/Lua/entity_map.lua:
```diff
@@ -65,12 +65,12 @@ the assertion that the entity is contained in the map to fail.
+function EntityMap:removeEntity(x, y, entity)
   -- Iterates through a table and removes the given entity
-  local function remove_entity_from_table(entity, entity_table)
+  local function remove_entity_from_table(enty, entity_table)
```
Promote this local function to top level, same reasons as the previous one.

TheCycoONE commented on CorsixTH/Lua/epidemic.lua:
```diff
@@ -599,11 +599,11 @@ end
   -- General usage tile finder, used in the other cases
   -- when the patient isn't sitting on a bench
-  local function get_best_usage_no_bench(nurse,patient)
+  local function get_best_usage_no_bench(nurse2, patient2)
```
Possibly promote the function instead. It does use the closure for self, so the entity map would have to be passed in another way.

TheCycoONE commented on CorsixTH/Lua/game_ui.lua:
```diff
@@ -459,25 +459,25 @@ function GameUI:onMouseMove(x, y, dx, dy)
        y >= self.app.config.height - scroll_region_size) then
-    local dx = 0
-    local dy = 0
+    local dx2 = 0
```
May have noticed I'm not a fan of *2 variables. Here dx, dy are change in the x / y location of the mouse between calls of ```onMouseMove``` dx2 and dy2 are the scroll distances. Silly things are done with ```tick_scroll_amount_mouse``` otherwise I'd suggest dropping the temp variable entirely. Suggesting ```tsam_x``` and ```tsam_y``` instead.

TheCycoONE commented on CorsixTH/Lua/graphics.lua:
```diff
@@ -359,8 +359,8 @@ function Graphics:loadLanguageFont(name, sprite_table, ...)
       font:setSheet(sprite_table)
-      local function reloader(font)
-        font:clearCache()
+      local function reloader(fnt)
```
Could move this function to top level and rename it ```font_reloader```

TheCycoONE commented on CorsixTH/Lua/dialogs/bottom_panel.lua:
```diff
@@ -721,7 +721,7 @@ function UIBottomPanel:addDialog(dialog_class, extra_function)
         local dialog = _G[dialog_class](self.ui)
         if extra_function then
-          _G[dialog_class][extra_function](d)
+          _G[dialog_class][extra_function](dialog)
```
Incidentally the only place this is used is to call ```UIBankManager.showStatistics``` https://github.com/CorsixTH/CorsixTH/blob/master/CorsixTH/Lua/dialogs/fullscreen/bank_manager.lua#L318

The argument to that function is a bool for whether to replace the cursor.

TheCycoONE replied:
dialog is right. With this notation self needs to be explicit.

Alberth289346 commented:
Loooong patch queue, but it addresses all points raised by lua check.
Many thanks to bobcanthelpyou for the initial start of the patch in #972 

Things are split by file, and while it's a lot of patches, they seem quite readable.

Alberth289346 commented on CorsixTH/Lua/entities/patient.lua:
```diff
@@ -199,7 +199,7 @@ function Patient:completeDiagnosticStep(room)
     local attn_detail = room.staff_member.profile.attention_to_detail / divisor
     local skill = room.staff_member.profile.skill / divisor
-    local diagnosis_bonus = (attn_detail + 0.4) *  skill
+    diagnosis_bonus = (attn_detail + 0.4) * skill
```
This is a bugfix

TheCycoONE commented:
For future consideration splitting by file isn't very helpful, it is pretty easy to see the file boundaries in a commit, and it makes navigation here more cumbersome.

Alberth289346 commented:
This isn't working

TheCycoONE commented:
I like your tags

Alberth289346 commented:
Merging this in smaller steps instead
